polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -107,14 +107,42 @@ impl<H: Copy> From<CandidateDescriptorV2<H>> for CandidateDescriptor<H> {
 	}
 }
 
-#[cfg(any(feature = "runtime-benchmarks", feature = "test"))]
-impl<H: Encode + Decode + Copy> From<CandidateDescriptor<H>> for CandidateDescriptorV2<H> {
+fn clone_into_array<A, T>(slice: &[T]) -> A
+where
+	A: Default + AsMut<[T]>,
+	T: Clone,
+{
+	let mut a = A::default();
+	<A as AsMut<[T]>>::as_mut(&mut a).clone_from_slice(slice);
+	a
+}
+
+impl<H: Copy> From<CandidateDescriptor<H>> for CandidateDescriptorV2<H> {
```

Comment:
what's the difference between this and the previous implementation?
is it just for performance?

Reply:
Yes, initially for testing it was good with encode/decode. At this point I am not sure if it is faster, but in theory should be.